From d66f40aebd7e4f4a0866987370a983558632000a Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Thu, 2 May 2019 13:03:50 +0200
Subject: [PATCH] arm64: dt: fsl-imx8[dx,dxp,qxp]-tqma8[xd,xdp,xqp]-mba8xx: add
 pcie gen1 tree

helps with issues with higher speeds not working for PCIe switch

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm64/boot/dts/freescale/Makefile          |  3 +++
 .../fsl-imx8dx-tqma8xd-mba8xx-pcie-gen1.dts     | 16 ++++++++++++++++
 .../fsl-imx8dxp-tqma8xdp-mba8xx-pcie-gen1.dts   | 17 +++++++++++++++++
 .../fsl-imx8qxp-tqma8xqp-mba8xx-pcie-gen1.dts   | 16 ++++++++++++++++
 .../dts/freescale/mba8xx-common-pcie-gen1.dtsi  | 17 +++++++++++++++++
 5 files changed, 69 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8xd-mba8xx-pcie-gen1.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8xdp-mba8xx-pcie-gen1.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8xqp-mba8xx-pcie-gen1.dts
 create mode 100644 arch/arm64/boot/dts/freescale/mba8xx-common-pcie-gen1.dtsi

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 3de8cbe2340b..5b6dd5277332 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -91,13 +91,16 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8dx-lpddr4-arm2.dtb \
 				  fsl-imx8dx-tqma8xd-mba8xx.dtb \
 				  fsl-imx8dx-tqma8xd-mba8xx-lvds-tm070jvhg33.dtb \
+				  fsl-imx8dx-tqma8xd-mba8xx-pcie-gen1.dtb \
 				  fsl-imx8dx-tqma8dxs-mb-smarc-2.dtb \
 				  fsl-imx8dxp-lpddr4-arm2.dtb \
 				  fsl-imx8dxp-tqma8xdp-mba8xx.dtb \
 				  fsl-imx8dxp-tqma8xdp-mba8xx-lvds-tm070jvhg33.dtb \
+				  fsl-imx8dxp-tqma8xdp-mba8xx-pcie-gen1.dtb \
 				  fsl-imx8dxp-tqma8dxps-mb-smarc-2.dtb \
 				  fsl-imx8qxp-tqma8xqp-mba8xx.dtb \
 				  fsl-imx8qxp-tqma8xqp-mba8xx-lvds-tm070jvhg33.dtb \
+				  fsl-imx8qxp-tqma8xqp-mba8xx-pcie-gen1.dtb \
 				  fsl-imx8qxp-tqma8xqp-mba8xx-dsilvds.dts \
 				  fsl-imx8qxp-tqma8qxps-mb-smarc-2.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8xd-mba8xx-pcie-gen1.dts b/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8xd-mba8xx-pcie-gen1.dts
new file mode 100644
index 000000000000..06ac5dc9eaee
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8xd-mba8xx-pcie-gen1.dts
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8dx-tqma8xd-mba8xx.dts"
+#include "mba8xx-common-pcie-gen1.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8xdp-mba8xx-pcie-gen1.dts b/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8xdp-mba8xx-pcie-gen1.dts
new file mode 100644
index 000000000000..fe9bc2708b44
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8xdp-mba8xx-pcie-gen1.dts
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+
+#include "fsl-imx8dxp-tqma8xdp-mba8xx.dts"
+#include "mba8xx-common-pcie-gen1.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8xqp-mba8xx-pcie-gen1.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8xqp-mba8xx-pcie-gen1.dts
new file mode 100644
index 000000000000..8221cf7e7ecf
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8xqp-mba8xx-pcie-gen1.dts
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2018 - 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qxp-tqma8xqp-mba8xx.dts"
+#include "mba8xx-common-pcie-gen1.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/mba8xx-common-pcie-gen1.dtsi b/arch/arm64/boot/dts/freescale/mba8xx-common-pcie-gen1.dtsi
new file mode 100644
index 000000000000..dd98f02441fa
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/mba8xx-common-pcie-gen1.dtsi
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+&pcieb{
+	fsl,max-link-speed = <1>;
+};
